v2/kafka-to-pubsub/src/main/java/com/google/cloud/teleport/v2/kafka/consumer/SslConsumerFactoryFn.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.security.auth.SecurityProtocol;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class to create Kafka Consumer with configured SSL.
+ */
+public class SslConsumerFactoryFn
+        implements SerializableFunction<Map<String, Object>, Consumer<byte[], byte[]>>
+{
+    private final Map<String, String> sslConfig;
+    private static final String TRUSTSTORE_LOCAL_PATH = "/tmp/kafka.truststore.jks";
+    private static final String KEYSTORE_LOCAL_PATH = "/tmp/kafka.keystore.jks";
+
+    /* Logger for class.*/
+    private static final Logger LOG = LoggerFactory.getLogger(SslConsumerFactoryFn.class);
+
+    public SslConsumerFactoryFn(Map<String, String> sslConfig) {
+        this.sslConfig = sslConfig;
+    }
+
+    @Override
+    public Consumer<byte[], byte[]> apply(Map<String, Object> config)
+    {
+        String bucket = sslConfig.get("bucket");
+        String trustStorePath = sslConfig.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG);
+        String keyStorePath = sslConfig.get(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG);
+        String trustStorePassword = sslConfig.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG);
+        String keyStorePassword = sslConfig.get(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG);
+        String keyPassword = sslConfig.get(SslConfigs.SSL_KEY_PASSWORD_CONFIG);
+        String outputTrustStoreFilePath;
+        String outputKeyStoreFilePath;
+        try
+        {
+            outputTrustStoreFilePath = TRUSTSTORE_LOCAL_PATH;
+            outputKeyStoreFilePath = KEYSTORE_LOCAL_PATH;
+            getGcsFileAsLocal(bucket, trustStorePath, outputTrustStoreFilePath);
+            getGcsFileAsLocal(bucket, keyStorePath, outputKeyStoreFilePath);
+        } catch (IOException e) {
+            LOG.error("Failed to retrieve data for SSL", e);
+            return new KafkaConsumer<>(config);
+        }
+
+        config.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, SecurityProtocol.SASL_SSL.name());
+        config.put(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG, outputTrustStoreFilePath);
+        config.put(SslConfigs.SSL_KEYSTORE_LOCATION_CONFIG, outputKeyStoreFilePath);
+        config.put(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG, trustStorePassword);
+        config.put(SslConfigs.SSL_KEYSTORE_PASSWORD_CONFIG, keyStorePassword);
+        config.put(SslConfigs.SSL_KEY_PASSWORD_CONFIG, keyPassword);
+
+        return new KafkaConsumer<>(config);
+    }
+
+    /**
+     * Reads a file from GCS and writes it locally.
+     *
+     * @param bucket GCS bucket name
+     * @param filePath path to file in GCS
+     * @param outputFilePath path where to save file locally
+     * @throws IOException thrown if not able to read or write file
+     */
+    public static void getGcsFileAsLocal(String bucket, String filePath, String outputFilePath) throws IOException {
```

Comment:
Code can be simplified as follows without depending on any external packages (org.apache.commons.io)
```
 import java.nio.file.Paths;
 import java.nio.file.StandardOpenOption;
import org.apache.beam.sdk.io.FileSystems;
import java.nio.channels.FileChannel;
import java.nio.channels.ReadableByteChannel;
public static void getGcsFileAsLocal(String bucket, String filePath, String outputFilePath) throws IOException {
       String gcsFilePath = String.format("gs://%s/%s", bucket, filePath);
        LOG.info("Reading contents from GCS file: {}", gcsFilePath);
        Set<StandardOpenOption> options = new HashSet<>(2);
        options.add(StandardOpenOption.CREATE);
        options.add(StandardOpenOption.APPEND);
        // Copy the GCS file into a local file and will throw an I/O exception in case file not found.
        try (ReadableByteChannel readerChannel = FileSystems.open(FileSystems.matchSingleFileSpec(gcsFilePath).resourceId())) { 
            try (FileChannel writeChannel = FileChannel.open(Paths.get(outputFilePath), options)) {
                writeChannel.transferFrom(readerChannel,0, Long.MAX_VALUE);
            }
        }
}

```

Reply:
Thanks! Applied your suggestion.